### PR TITLE
Adds CICS Resource builder build capability to zAppbuild

### PR DESCRIPTION
**Pull request by odttlnt**
This request extends the functionality of the build by allowing users to build CICS resource definitions in yaml format. The changes will allow the CICS yaml formatted file to be build by the CICS Resource builder tool. It will generate an artefact which will be in the legacy CSD format. This artefact can then be used in a DFHCSDUP job to apply the changes to the CICS region. The PR is a part of the Infrastructure as code effort.
The PR consists of CRB. groovy file for doing the build and its corresponding properties files CRB.properties. Plus related changes to the README

### CODE REVIEW

**Comment by M-DLB** [comment]
@odttlnt - Can you please take the latest version of CRB.properties from our internal reference system?
Also you don't need to change the property files for the Mortgage Application.
Thank you!

@dennis-behm - Can you please have a look for the rest of the changes? Thanks!

File: languages/CRB.groovy [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+
+@Field BuildProperties props = BuildProperties.getInstance()
+@Field def buildUtils= loadScript(new File("${props.zAppBuildDir}/utilities/BuildUtilities.groovy"))
+
+// verify required build properties
+buildUtils.assertBuildProperties(props.crb_requiredBuildProperties)
+
+List<String> buildList = argMap.buildList
+
+println("** Processing files mapped to ${this.class.getName()}.groovy script")
```
Comment (dennis-behm) [inline languages/CRB.groovy:29]: With the next release of zAppBuild, the language script have a progress indicator. You can find it already in the  `develop` branch contributed with https://github.com/IBM/dbb-zappbuild/commit/04e6118030f1764bd82f637223f6fdcbfac52ce9. Can we please have the same logic here?

**Review by dennis-behm — CHANGES_REQUESTED** [review]
Thank you for the contribution.

@odttlnt I have a few comments on this PR, please see them attached to the PR.

@odttlnt, @M-DLB, I am wondering, if the `crb_resourceModelFile` and `crb_applicationConstraintsFile` should be defined the `build-conf` directory of zAppBuild, rather then pointing to an "external" configuration location.

File: languages/CRB.groovy [hunk 2]
```diff
@@ -0,0 +1,90 @@
+@groovy.transform.BaseScript com.ibm.dbb.groovy.ScriptLoader baseScript
+import com.ibm.dbb.metadata.*
+import com.ibm.dbb.dependency.*
+import com.ibm.dbb.build.*
+import com.ibm.dbb.build.report.records.*
+import com.ibm.dbb.build.report.*
+import groovy.transform.*
+
+/*
```
[the remark this answers is not in the document]
Comment (odttlnt) [reply languages/CRB.groovy:9]: Have added this to the comments.
The script will be used to process CICS definition files from the repo with suffix .yml or .yaml.
The script will need the CICS definition file in YAML format, it will also need the Model file and 
the application constraints file.
The YAML formatted CICS definition files are created by the CICS TS resource builder tool.
Further information on the CICS TS resource builder tool and can be found at 
https://www.ibm.com/docs/en/cics-resource-builder/1.0?topic=overview
The link also gives a good explanation of the Model file and application constraints file and its usage.

File: languages/CRB.groovy [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+    def slashIndex = buildFile.lastIndexOf('/')
+    if (slashIndex < 0) slashIndex = 0
+	def outputFile = buildFile.substring(slashIndex + 1, extIndex) + ".csd"
+
+    println("*** Output file is ${props.buildOutDir}/$outputFile.")
+    // Build the shell command to execute
+    def applicationParm = ""
+    if (applicationConstraintsFile) 
+        applicationParm = "--application $applicationConstraintsFile"
+    def zrb_cmd = zrbPath + " build --model $resourceModelFile $applicationParm --resources ${props.workspace}/${buildFile} --output ${props.buildOutDir}/$outputFile"
```
[the remark this answers is not in the document]
Comment (odttlnt) [reply languages/CRB.groovy:75]: Added
if (props.verbose)
        println("*** Executing zrb command: $zrb_cmd")

File: languages/CRB.groovy (hunk 1 above)
Comment (odttlnt) [reply languages/CRB.groovy]: Changed to show progress indicator

File: samples/MortgageApplication/application-conf/application.properties [hunk 4]
```diff
@@ -3,7 +3,7 @@
 #
 # Comma separated list of additional application property files to load
 # Supports both relative path (to ${application}/application-conf/) and absolute path
-applicationPropFiles=file.properties,BMS.properties,Cobol.properties,LinkEdit.properties,languageConfigurationMapping.properties
+applicationPropFiles=file.properties,BMS.properties,Cobol.properties,LinkEdit.properties,languageConfigurationMapping.properties,CRB.properties
```
[the remark this answers is not in the document]
Comment (odttlnt) [reply samples/MortgageApplication/application-conf/application.properties]: Removed.

File: samples/application-conf/CRB.properties [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+# Cics Resource Builder path 
+crb_zrbLocation=/var/crb-1.0.3/cics-resource-builder/bin/zrb
+
+#
+# Cics Resource builder max acceptable exit/return code
+crb_maxRC=4
+
+#
+# CRB model yaml file
+crb_resourceModelFile=/var/crb-1.0.3/resourcesModel.yml
```
[the remark this answers is not in the document]
Comment (odttlnt) [reply samples/application-conf/CRB.properties]: Added for both Model file and Application constraints

**Comment by M-DLB** [comment]
@dennis-behm The challenge is that ```crb_resourceModelFile``` and ```crb_applicationConstraintsFile``` are application-related (different for each application) but are provided by CICS admins, so they are not supposed to be pat of the application's Git repo. We thought providing an override with an absolute path would be the best way.

File: samples/MortgageApplication/application-conf/file.properties [hunk 6]
```diff
@@ -7,6 +7,7 @@ dbb.scriptMapping = BMS.groovy :: **/*.bms
 dbb.scriptMapping = Cobol.groovy :: **/*.cbl
 dbb.scriptMapping = LinkEdit.groovy :: **/*.lnk
 dbb.scriptMapping = PLI.groovy :: **/*.pli
+dbb.scriptMapping = CRB.groovy :: **/*.yml, **/*.yaml
```
Comment (dennis-behm) [inline samples/MortgageApplication/application-conf/file.properties:10]: Please remove the scriptMapping from the MortgageApplication sample.

File: languages/CRB.groovy [hunk 7]
```diff
@@ -0,0 +1,105 @@
+@groovy.transform.BaseScript com.ibm.dbb.groovy.ScriptLoader baseScript
+import com.ibm.dbb.metadata.*
+import com.ibm.dbb.dependency.*
+import com.ibm.dbb.build.*
+import com.ibm.dbb.build.report.records.*
+import com.ibm.dbb.build.report.*
+import groovy.transform.*
+
+/*
+The language script is for running the CICS tool (zrb) on CICS definitions in YML (yaml)
```
Comment (dennis-behm) [inline languages/CRB.groovy:10]: A minor comment.

The _CICS tool_ is a confusing term for me. Instead of 
```
The language script is for running the CICS tool (zrb) on CICS definitions in YML (yaml)
 format ... 
```
can we please rephrase it to:
```
This language script is invoking the 'CICS TS resource builder utility' for the CICS resource definitions defined in the YAML format.

The script processes CICS definition files stored in the git repository with suffix .yml or .yaml to produce a CSD formatted file.

The model file and the application constraints file, required by the CICS TS resource builder, are referenced via the crb.properties files. 

The script expects that the CICS TS resource builder is installed on the build machine in z/OS UNIX. The location of it is also referenced in the properties files.

The output of the process is a CSD formatted file, which is reported in the DBB Build report for further post-build processing such as packaging and deployment into a target environment.

Further information on the CICS TS resource builder utility and can be found at 
 https://www.ibm.com/docs/en/cics-resource-builder/1.0?topic=overview

```

File: samples/application-conf/CRB.properties [hunk 8]
```diff
@@ -0,0 +1,27 @@
+# Language properties used by language/CRB.groovy
```
Comment (dennis-behm) [inline samples/application-conf/CRB.properties:1]: @odttlnt @M-DLB ,
I see that these definitions are part of `application-conf` and `build-conf`. What would be the best location to manage these property definitions? Can we propose just one? 

If necessary, we would need to split it up into central build config, like `crb_zrbLocation` in build-conf and application specifics into the application configuration dir.

**Review by dennis-behm — CHANGES_REQUESTED** [review]
Couple of additional comments.

File: languages/CRB.groovy (hunk 7 above)
Comment (odttlnt) [reply languages/CRB.groovy:10]: Updated as requested

File: samples/MortgageApplication/application-conf/file.properties (hunk 6 above)
Comment (odttlnt) [reply samples/MortgageApplication/application-conf/file.properties:10]: Fixed

Comment (odttlnt) [reply samples/MortgageApplication/application-conf/file.properties]: Fixed

File: languages/CRB.groovy (hunk 7 above)
Comment (odttlnt) [reply languages/CRB.groovy]: Changed as requested

File: samples/application-conf/CRB.properties (hunk 8 above)
Comment (odttlnt) [reply samples/application-conf/CRB.properties:1]: I think it should be as is. Primarily because it provides more flexibility to the developer. They can mix and match the parameters and values

**Review by dennis-behm — APPROVED** [review]

**Comment by dennis-behm** [comment]
Thank you for the updates @odttlnt .  This PR is depending on #308 , because YAML files should not be scanned with the default DBB dependency scanner.

FYI - We had some un-signed commits on `develop` why I had to rebase the last 3 commits. The PR looks corrupted, but we will take care during the merge process and squash them into a new one.

**Comment by dennis-behm** [comment]
Superseded by #391 -  because this PR is based on an outdated version of zAppBuild
